GVRf/Framework/src/org/gearvrf/GVRRenderTexture.java

```diff
@@ -55,6 +55,20 @@ public GVRRenderTexture(GVRContext gvrContext, int width, int height,
     GVRRenderTexture(GVRContext gvrContext, long ptr) {
         super(gvrContext, ptr);
     }
+    
+    /**
+     * Return the width of GVRRenderTexture (FBO)
+     */
+    public int getWidth() {
+    	return NativeRenderTexture.getWidth(getPtr());
+    }
+
```

Comment:
Fair enough, but I think it may be giving too much deference to the original ‘design’. I can’t see a lot of point to adding so much code – and burning so many cycles – to ask the jni side for data that came from Java in the first place.

From: Guodong Rong [mailto:notifications@github.com]
Sent: Monday, April 20, 2015 5:00 PM
To: Samsung/GearVRf
Cc: Jon Shemitz
Subject: Re: [GearVRf] Add 2D screenshot APIs for left eye, right eye, and center eye. (#15)


In GVRf/Framework/src/org/gearvrf/GVRRenderTexture.java<https://github.com/Samsung/GearVRf/pull/15#discussion_r28740252>:

> @@ -55,6 +55,20 @@ public GVRRenderTexture(GVRContext gvrContext, int width, int height,

>      GVRRenderTexture(GVRContext gvrContext, long ptr) {

>          super(gvrContext, ptr);

>      }

> +

> +    /**

> +     * Return the width of GVRRenderTexture (FBO)

> +     */

> +    public int getWidth() {

> +     return NativeRenderTexture.getWidth(getPtr());

> +    }

> +

My understanding is that GVRF is designed in the way that we do not store anything in Java side and everything is stored in native side. So we should get information from native side when needed.

This is the reason I implemented them in this "wired" way.

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/15/files#r28740252>.